api/src/lib.rs

```diff
@@ -22,6 +22,49 @@ pub fn to_png(image: &[u8]) -> Result<Vec<u8>, JsValue> {
     read_and_encode(image, |img| img)
 }
 
+#[wasm_bindgen]
+pub fn remove_hot_pixels_with_percentile(
+    image_data: &[u8],
+    percentile: f32,
+) -> Result<Vec<u8>, JsValue> {
+    let img = image::load_from_memory(image_data)
+        .map_err(|e| JsValue::from_str(&format!("Image decode error: {}", e)))?;
+
+    let rgb = img.to_rgb8();
+    let (width, height) = rgb.dimensions();
+    let mut pixels = rgb.into_raw();
+
+    let total = pixels.len();
+    if total == 0 {
+        return Err(JsValue::from_str("Empty image buffer"));
+    }
+    if !(0.0..=100.0).contains(&percentile) {
+        return Err(JsValue::from_str("Percentile must be between 0 and 100"));
+    }
```

Comment:
I would also move it to the beginning of this function because this check does not depend on image transformations.
```suggestion
    if percentile < 0.0 || percentile > 100.0 {
        return Err(JsValue::from_str("Percentile must be between 0 and 100"));
    }
```